src/shmem_team.c

```diff
@@ -0,0 +1,460 @@
+/* -*- C -*-
+ *
+ * Copyright (c) 2019 Intel Corporation. All rights reserved.
+ * This software is available to you under the BSD license.
+ *
+ * This file is part of the Sandia OpenSHMEM software package. For license
+ * information, see the LICENSE file in the top level directory of the
+ * distribution.
+ *
+ */
+
+
+#include "config.h"
+
+#include "shmemx.h"
+#include "shmem_team.h"
+#include "shmem_collectives.h"
+#include "shmem_remote_pointer.h"
+
+#include <math.h>
+
+#define SHMEMX_TEAM_WORLD_INDEX   0
+#define SHMEMX_TEAM_SHARED_INDEX  1
+
+shmem_internal_team_t shmem_internal_team_world;
+shmemx_team_t SHMEMX_TEAM_WORLD = (shmemx_team_t) &shmem_internal_team_world;
+
+shmem_internal_team_t shmem_internal_team_shared;
+shmemx_team_t SHMEMX_TEAM_SHARED = (shmemx_team_t) &shmem_internal_team_shared;
+
+shmem_internal_team_t **shmem_internal_team_pool;
+long *shmem_internal_psync_pool;
+long *shmem_internal_psync_barrier_pool;
+static unsigned char *psync_pool_avail;
+static unsigned char *psync_pool_avail_reduced;
+
+shmem_internal_team_op_t shmem_internal_team_sync_type     = SYNC;
+shmem_internal_team_op_t shmem_internal_team_bcast_type    = BCAST;
+shmem_internal_team_op_t shmem_internal_team_reduce_type   = REDUCE;
+shmem_internal_team_op_t shmem_internal_team_collect_type  = COLLECT;
+shmem_internal_team_op_t shmem_internal_team_alltoall_type = ALLTOALL;
```

Comment:
Noted in the header, the `shmem_internal_team_*_type` constants appear to be redundant.  Better to use the enumeration constants directly.